pages/login/LoginHandler.php

```diff
@@ -484,4 +506,37 @@ protected function sendHome($request)
             $request->redirect(null, 'user');
         }
     }
+
+    /**
+     * Validate if ALTCHA user's response is valid
+     *
+     * @param PKPRequest $request
+     * @param string $altchaConfigKey the key to search on config.inc.php
+     */
+    private function validateAltchaResponse($request, $altchaConfigKey): ?string
+    {
+        if (Config::getVar('captcha', 'altcha') && Config::getVar('captcha', $altchaConfigKey)) {
+            try {
+                FormValidatorAltcha::validateResponse($request->getUserVar('altcha'), $request->getRemoteAddr());
+                return null;
+            } catch (Exception $exception) {
+                return 'common.captcha.error.missing-input-response';
+            }
+        }
+    }
+
+    /**
+     * Generate ALTCHA challenge to use it on form in case ALTCHA
+     * is enabled on the specific page
+     *
+     * @param string $altchaConfigKey the key to search on config.inc.php
+     * @param TemplateManager $templateMgr
+     */
+    private function generateAltchaComponent($altchaConfigKey, &$templateMgr): void
```

Comment:
Wherever possible, add typehints in new code; also remove redundant code self-documentation while you're at it. The `@param string $altchaConfigKey` adds a description, so it's valuable, but once it has a typehint, `@param TemplateManager $templateMgr` adds nothing and should be removed.